gammapy/modeling/parameter.py

```diff
@@ -247,6 +247,11 @@ def copy(self):
         """A deep copy"""
         return copy.deepcopy(self)
 
+    def update_from_dict(self, data):
+        """Update parameters from a dict.
+            Key name is accessed as a hidden parameters with _keyname. """
+        self.__dict__.update({ '_'+k: v for k, v in data.items() })
```

Comment:
Maybe we should forbid (or ignore) to modify the `.name` attribute, because I think it will break model evaluation...

Reply:
indeed. It could be ignored. There is also an issue with min, max if set to 'nan' this is a string instead of a float. in dict or table, it shows as nan but it is np.nan in reality.

Reply:
I think this is handled in the min and max setters: https://github.com/gammapy/gammapy/blob/master/gammapy/modeling/parameter.py#L169

I now realise going through `__dict__` was maybe a bad idea..because we do some type casting / checking in the setters. However using `setattr()` would probably work better:
```python3
from gammapy.modeling import Parameter
p = Parameter("test", value=4)
setattr(p, "min", "42")
print(p.min)
```

Reply:
indeed, this can be problematic with value for example with factor and scaling

Reply:
Maybe change the implementation to rely on on `setattr(self, k, v)`, then it's safe?

Reply:
changed to `setattr()` and it indeed solves the issues when handling values and units, etc
Also simpler to read as there is no need to access hidden param.